### DISCUSSION TITLE
PlatformIO and Arduino Uno - it wants to upload to many other boards?

### DISCUSSION THREAD

**Discussion by hmmueller** [Paciente8159/uCNC#710] [Q&A] [answered]
Ok - I have (and have been) converted to VSCode and PlatformIO - one hell of installation madness, but so it goes in these modern times.

So now I am on the master branch (for the moment). It compiles, it does an upload, it says 

> avrdude: 31862 bytes of flash verified
> avrdude: safemode: Fuses OK (E:00, H:00, L:00)
> avrdude done.  Thank you.
> ========================== [SUCCESS] Took 26.05 seconds ==========================

... really nice (especially the "Thank you" in the middle: What did I do that it thanks me? ;-) ).

But then all hell breaks loose: It wants to do 

> Processing AVR-MEGA2560-RAMPS-V1_4

and runs into

> Uploading .pio\build\AVR-MEGA2560-RAMPS-V1_4\firmware.hex
> avrdude: stk500v2_ReceiveMessage(): timeout

still it says :-)

> avrdude done.  Thank you.

but then

> =========================== [FAILED] Took 69.11 seconds ===========================

Then

> Processing AVR-RAMBO (board: megaatmega2560 
> ...
> =========================== [FAILED] Took 67.79 seconds ===========================

and

> Processing SAMD21-Wemos-M0
> ...
> =========================== [FAILED] Took 16.57 seconds ===========================

and finally

> Processing STM32F1-Bluepill-F103C8
with 
> MissingPackageManifestError: Could not find one of 'package.json' manifest files in the package

What's going on here? How can I suppress these attempts for platforms (I assume) that I dont have (and dont know about)?

Thank you! (Yes, I can also be polite, like PlatformIO :-))
H.M.

**Comment by hmmueller** [comment]
My own attempt at an answer: I modified platformio.ini as follows:

```
default_envs = AVR-UNO
; , AVR-MEGA2560-RAMPS-V1_4, AVR-RAMBO, SAMD21-Wemos-M0, STM32F0-Bluepill-F030C8, STM32F1-Bluepill-F103C8, STM32F4-Blackpill-F401CC, ESP8266-Wemos-D1-R2, LPC176X-RE-ARM, ESP32-Wemos-D1-R32, RP2040-PICO-W, RP2040-PICO-W-MULTICORE
extra_configs = 
  uCNC/src/hal/boards/avr/avr.ini
;  uCNC/src/hal/boards/samd21/samd21.ini
;  uCNC/src/hal/boards/stm32/stm32.ini
;  uCNC/src/hal/boards/esp8266/esp8266.ini
;  uCNC/src/hal/boards/lpc176x/lpc176x.ini
;  uCNC/src/hal/boards/esp32/esp32.ini
;  uCNC/src/hal/boards/rp2040/rp2040.ini
  uCNC/src/hal/mcus/virtual/virtual.ini

```
i.e., everything exception AVR-UNO and avr.ini is commented out. Seems to do what I want. If this is ok, no answer is needed - but if this destroys something I'd need along the way, please drop me a line.

Thanks!
H.M.

**Comment by Paciente8159** [comment, marked as answer]
you don't need to do any of that.
The problem is that you are building the default environment, that essentially builds the firmware for all boards.

You need to select your target environment from the PlatformIO tool bar as well as you board COM port

![image](https://github.com/user-attachments/assets/b8e48b57-ae25-468a-aadf-cfd1d56226b9)

**Comment by hmmueller** [comment]
Thanks! - still on the learning curve. Will do that tomorrow.Good night!H.M.--Gesendet mit der GMX Mail AppAm 28.07.24, 21:28 schrieb Paciente8159 ***@***.***>:

   
  you don't need to do any of that. The problem is that you are building the default environment, that essentially builds the firmware for all boards. 
  You need to select your target environment from the PlatformIO tool bar as well as you board COM port 
  image.png (view on web)
